test/lib/jdk/test/lib/compiler/InMemoryJavaCompiler.java

```diff
@@ -162,6 +169,86 @@ public boolean hasLocation(Location location) {
 
     }
 
+    // Wraper for class file
+    static class ClassFile extends SimpleJavaFileObject {
+
+        private final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        protected ClassFile(String name) {
+            super(URI.create("memo:///" + name.replace('.', '/') + Kind.CLASS.extension), Kind.CLASS);
+        }
+
+        @Override
+        public ByteArrayOutputStream openOutputStream() { return this.baos; }
+
+        byte[] toByteArray() { return baos.toByteArray(); }
+    }
+
+    // File manager which spawns ClassFile instances by demand
+    static class FileManager extends ForwardingJavaFileManager<JavaFileManager> {
+
+        private Map<String, ClassFile> classesMap = new HashMap<String, ClassFile>();
+
+        protected FileManager(JavaFileManager fileManager) {
+            super(fileManager);
+        }
+
+        @Override
+        public ClassFile getJavaFileForOutput(Location location, String name, JavaFileObject.Kind kind, FileObject source) {
+            ClassFile classFile = new ClassFile(name);
+            classesMap.put(name, classFile);
+            return classFile;
+        }
+
+        public Map<String, byte[]> getByteCode() {
+            Map<String, byte[]> result = new HashMap<String, byte[]>();
+            for (Entry<String, ClassFile> entry : classesMap.entrySet()) {
+                result.put(entry.getKey(), entry.getValue().toByteArray());
+            }
+            return result;
+        }
+    }
+
+    // Wrapper for source file
+    static class SourceFile extends SimpleJavaFileObject {
+
+        private CharSequence sourceCode;
+
+        public SourceFile(String name, CharSequence sourceCode) {
+            super(URI.create("memo:///" + name.replace('.', '/') + Kind.SOURCE.extension), Kind.SOURCE);
+            this.sourceCode = sourceCode;
+        }
+
+        @Override
+        public CharSequence getCharContent(boolean ignore) {
+            return this.sourceCode;
+        }
+    }
+
+    public static Map<String, byte[]> compile(Map<String, ? extends CharSequence> inputMap) {
```

Comment:
Could the tests that use this 'compile' be easily fixed to use the below version of 'compile' so you can delete this too?

Reply:
Agree, most of tests should be updated to use compile(String className ...)
and remove wrapping the single file into Map.

While there are tests that need this method to compile several source files as a single batch.

This comments should be added to this method , and it might be renamed to clearly says that it compile a batch of files.